Building with 'platform=x11' fails in spite of alias to 'linuxbsd'

**Godot version:**
Master branch
#37153 

Godot Engine v4.0.dev.custom_build.8dd58a615

**OS/device including version:**
Linux 5.5.13-arch1-1 x86_64

**Steps to reproduce:**
Run the command:
`scons -j8 platform=x11`

**Issue description:**
Specifying the command below:

`scons -j8 platform=x11`

Produces the build error:
```
thirdparty/vulkan/loader/loader.c:81:10: fatal error: loader_cmake_config.h: No such file or directory
   81 | #include "loader_cmake_config.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
scons: *** [thirdparty/vulkan/loader/loader.linuxbsd.tools.64.o] Error 1
Ranlib Library         ==> modules/libmodule_assimp.linuxbsd.tools.64.a
scons: building terminated because of errors.
```
If we are to use `platform=linuxbsd` then there are no problems with building.

The error can be traced to `loader_cmake_config.h` needing `VULKAN_NON_CMAKE_BUILD` to be defined. The setting of this can then be traced to the file `drivers/vulkan/SCsub`.
```
 elif env["platform"] == "linuxbsd":
        env_thirdparty.AppendUnique(
            CPPDEFINES=[
                "VK_USE_PLATFORM_XLIB_KHR",
                "VULKAN_NON_CMAKE_BUILD",
                'SYSCONFDIR=\\"%s\\"' % "/etc",
                'FALLBACK_DATA_DIRS=\\"%s\\"' % "/usr/local/share:/usr/share",
                'FALLBACK_CONFIG_DIRS=\\"%s\\"' % "/etc/xdg",
            ]
        )
```
There is a commit #37369 that made an alias that turns `platform=x11` to `platform=linuxbsd`. Unfortunately, this alias does not seem to carry over to `SCsub`.

**Possible fix:**
Adding checks for the other aliases in #37369 in `SCsub` is proven to fix the problem.

```
elif env["platform"] == "linuxbsd" or env["platform"] == "x11" or env["platform"] == "linux" or env["platform"] == "bsd":
```
I am open to fixing this and making a pull request. However, I am unsure if the possible fix above is the correct approach. Any help would be appreciated.

Thanks!

--- a/SConstruct
+++ b/SConstruct
@@ -258,6 +258,7 @@ if selected_platform in ["linux", "bsd", "x11"]:
         print('Platform "x11" has been renamed to "linuxbsd" in Godot 4.0. Building for platform "linuxbsd".')
     # Alias for convenience.
     selected_platform = "linuxbsd"
+    env_base["platform"] = selected_platform
 
 if selected_platform in platform_list:
     tmppath = "./platform/" + selected_platform

